lib/dialects/snowflake/query-generator.js

```diff
@@ -0,0 +1,633 @@
+    if (database) {
+      query += ` AND TABLE_SCHEMA = ${this.escape(database)}`;
+    } else {
+      query += ' AND TABLE_SCHEMA NOT IN ( \'INFORMATION_SCHEMA\', \'PERFORMANCE_SCHEMA\', \'SYS\')';
+    }
+    return `${query};`;
+  }
+
+  addColumnQuery(table, key, dataType) {
+    return Utils.joinSQLFragments([
+      'ALTER TABLE',
+      this.quoteTable(table),
+      'ADD',
+      this.quoteIdentifier(key),
+      this.attributeToSQL(dataType, {
+        context: 'addColumn',
+        tableName: table,
+        foreignKey: key
+      }),
+      ';'
+    ]);
+  }
+
+  removeColumnQuery(tableName, attributeName) {
+    return Utils.joinSQLFragments([
+      'ALTER TABLE',
+      this.quoteTable(tableName),
+      'DROP',
+      this.quoteIdentifier(attributeName),
+      ';'
+    ]);
+  }
+
+  changeColumnQuery(tableName, attributes) {
+    const query = subQuery => `ALTER TABLE ${this.quoteTable(tableName)} ALTER COLUMN ${subQuery};`;
+    const sql = [];
+    for (const attributeName in attributes) {
+      let definition = this.dataTypeMapping(tableName, attributeName, attributes[attributeName]);
+      let attrSql = '';
+
+      if (definition.includes('NOT NULL')) {
+        attrSql += query(`${this.quoteIdentifier(attributeName)} SET NOT NULL`);
+
+        definition = definition.replace('NOT NULL', '').trim();
+      } else if (!definition.includes('REFERENCES')) {
+        attrSql += query(`${this.quoteIdentifier(attributeName)} DROP NOT NULL`);
+      }
+
+      if (definition.includes('DEFAULT')) {
+        attrSql += query(`${this.quoteIdentifier(attributeName)} SET DEFAULT ${definition.match(/DEFAULT ([^;]+)/)[1]}`);
+
+        definition = definition.replace(/(DEFAULT[^;]+)/, '').trim();
+      } else if (!definition.includes('REFERENCES')) {
+        attrSql += query(`${this.quoteIdentifier(attributeName)} DROP DEFAULT`);
+      }
+
+      if (attributes[attributeName].startsWith('ENUM(')) {
+        attrSql += this.pgEnum(tableName, attributeName, attributes[attributeName]);
+        definition = definition.replace(/^ENUM\(.+\)/, this.pgEnumName(tableName, attributeName, { schema: false }));
+        definition += ` USING (${this.quoteIdentifier(attributeName)}::${this.pgEnumName(tableName, attributeName)})`;
+      }
+
+      if (definition.match(/UNIQUE;*$/)) {
+        definition = definition.replace(/UNIQUE;*$/, '');
+        attrSql += query(`ADD UNIQUE (${this.quoteIdentifier(attributeName)})`).replace('ALTER COLUMN', '');
+      }
+
+      if (definition.includes('REFERENCES')) {
+        definition = definition.replace(/.+?(?=REFERENCES)/, '');
+        attrSql += query(`ADD FOREIGN KEY (${this.quoteIdentifier(attributeName)}) ${definition}`).replace('ALTER COLUMN', '');
+      } else {
+        attrSql += query(`${this.quoteIdentifier(attributeName)} TYPE ${definition}`);
+      }
+
+      sql.push(attrSql);
+    }
+
+    return sql.join('');
```

Comment:
nit: use `Utils.joinSQLFragments`

Reply:
better to stay existing for now...the logic is different with `Utils.joinSQLFragments` with auto adding space

Reply:
This `changeColumnQuery` is close to the postgres when other part of are more close to mysql....

I will refactor it to use `joinSQLFragments`

Reply:
Done - use `joinSQLFragens` and also remove the ENUM code ( since snowflake doesn't support it )